crates/rome_formatter/src/jsx/tag/self_closing_element.rs

```diff
@@ -1,7 +1,26 @@
-use crate::{FormatElement, FormatResult, Formatter, ToFormatElement};
-use rome_js_syntax::{AstNode, JsxSelfClosingElement};
+use crate::formatter_traits::FormatTokenAndNode;
+use crate::group_elements;
+use crate::{
+    format_elements, soft_block_indent, soft_line_break_or_space, space_token, FormatElement,
+    FormatResult, Formatter, ToFormatElement,
+};
+use rome_js_syntax::JsxSelfClosingElement;
+
 impl ToFormatElement for JsxSelfClosingElement {
     fn to_format_element(&self, formatter: &Formatter) -> FormatResult<FormatElement> {
-        Ok(formatter.format_verbatim(self.syntax()))
+        let mut attributes = Vec::new();
+        for attribute in self.attributes() {
+            attributes.push(attribute.format(formatter)?);
+            attributes.push(soft_line_break_or_space());
+        }
```

Comment:
You can use `join_elements` for this together with `format_nodes`
```suggestion
        let attributes = join_elements(soft_line_break_or_space(), formatter.format_nodes(self.attributes());
```

Reply:
Hmm so I switched to this with a tweak:

```
let mut attributes = Vec::new();
for attribute in self.attributes() {
      attributes.push(attribute.format(formatter)?);
      attributes.push(soft_line_break_or_space());
}
```

But then for some reason the lines always became a hard line break. I don't really know why.

Reply:
If possible, we should avoid creating new vectors. `join_elements` uses `concat_elements` under the hoods, which  is really optimized around the usage of new vectors.